mteb/evaluation/evaluators/model_classes.py

```diff
@@ -0,0 +1,562 @@
+from __future__ import annotations
+
+import heapq
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Any
+
+import torch
+import tqdm
+from sentence_transformers import CrossEncoder, SentenceTransformer
+
+from mteb.encoder_interface import Encoder, PromptType
+from mteb.model_meta import ModelMeta
+
+from .utils import convert_conv_history_to_query, cos_sim, dot_score, download
+
+logger = logging.getLogger(__name__)
+
+
+def corpus_to_str(
+    corpus: list[dict[str, str]] | dict[str, list[str]] | list[str],
+) -> list[str]:
+    if isinstance(corpus, dict):
+        sentences = [
+            (corpus["title"][i] + " " + corpus["text"][i]).strip()
+            if "title" in corpus
+            else corpus["text"][i].strip()
+            for i in range(len(corpus["text"]))
+        ]
+    elif isinstance(corpus, list) and isinstance(corpus[0], dict):
+        sentences = [
+            (doc["title"] + " " + doc["text"]).strip()
+            if "title" in doc
+            else doc["text"].strip()
+            for doc in corpus
+        ]
+    else:
+        sentences = corpus
+    return sentences
+
+
+# Adapted from https://github.com/beir-cellar/beir/blob/f062f038c4bfd19a8ca942a9910b1e0d218759d4/beir/retrieval/search/dense/exact_search.py#L12
+class DenseRetrievalExactSearch:
+    def __init__(
+        self,
+        model: Encoder,
+        encode_kwargs: dict[str, Any] = {},
+        corpus_chunk_size: int = 50000,
+        previous_results: str | Path | None = None,
+        **kwargs: Any,
+    ):
+        # Model is class that provides encode_corpus() and encode_queries()
+        self.model = model
+        self.encode_kwargs = encode_kwargs
+
+        if "batch_size" not in encode_kwargs:
+            encode_kwargs["batch_size"] = 128
+        if "show_progress_bar" not in encode_kwargs:
+            encode_kwargs["show_progress_bar"] = True
+        if "convert_to_tensor" not in encode_kwargs:
+            encode_kwargs["convert_to_tensor"] = True
```

Comment:
I think this can be removed and `convert_to_tensor` can be passed explicitly to `encode`  https://github.com/embeddings-benchmark/mteb/issues/1325